### DISCUSSION TITLE
Improve instructions for cloning Extras into Anvil IDE

### DISCUSSION THREAD

**Discussion by hugetim** [anvilistas/anvil-extras#232] [Ideas]
1. Add instructions for connecting to GitHub with SSH to Enterprise Installation docs (See: https://anvil.works/forum/t/how-do-i-clone-anvil-extras/9601/10)
2. These instructions should also be linked to in the Contributing guide, to help contributors create a debugging (and manual testing) environment.

**Comment by rhurlbatt** [comment]
I am very much in favour of this. This is my first time working with Git and although I am very keen to contribute - there are a lot of stumbling blocks for new players. Eliminating these would ease the way for people to contribute - yes this does make it tougher since newbies like myself will probably produce some terrible code - but we all start somewhere right?

I am happy to do the writing from a newbies perspective but would be grateful for some guidance on the entire step by step process. Would anyone be keen to spend half an hour to walk me through this and I will document it from there? Fair warning- it may be painful to sit through 🙄

**Comment by s-cork** [comment]
@meatballs - updated some documentation about cloning the repo into the anvil IDE:
https://anvil-extras.readthedocs.io/en/latest/guides/installation.html#install-as-a-clone

I'd probably start by creating a github fork (you've already done this)
- Cloning this fork locally (it's now the `origin` remote)
- Creating a blank app in the anvil editor called `anvil_extras`
- Get the clone url (ignore the git clone part - just the url)
- use the command `git remote add anvil ssh/.......git` (it's now the `anvil` remote)
- `git push -f anvil main`
- go and check to see if the blank app is now fully populated.
- Create the branch you want to work on (either locally or in the anvil editor)
- Then start hacking away
- At some point you'll want to pull your changes locally - maybe this is the stage to pause if you get this far!


You will also want the anvil-extras branch added as a remote locally
`git remote add anvil-extras git@github.com:anvilistas/anvil-extras.git`
This is the source of truth
You will occasionally need to fetch changes from the anvil-extras branch and rebase your version of main onto the anvil-extras version of main.
In fact - you should never touch the main branch.
You should only ever fetch and rebase from anvil-extras.
All your changes should be on a different branch to main.


That was probably too much info!


It may be helpful to pick a local editor that helps with GIT.
I use VS code with git lens and git graph installed as plugins.
But most good editors will have their own plugins.
This helps me visualise the branches more easily.

**Reply by rhurlbatt** [reply]
This is fantastic thank you @s-cork

If there are no objections, I will turn the above in to some detailed docs intended to take a new contributor right the way through.
I have used VS code and Pycharm but have no preference. If there is a preferred local IDE that works well I will muddle my way through installing that.

**Comment by meatballs** [comment]
> Would anyone be keen to spend half an hour to walk me through this and I will document it from there?

I'd be happy to - but finding a slot will be tricky. It would have to be some time next week at the earliest.

**Reply by rhurlbatt** [reply]
Thanks Owen that would be great - I can work to your schedule as I know you are flat out.
Will PM you to sort a time that suits
